src/toast/hoc/toast.jsx

```diff
@@ -0,0 +1,18 @@
+import { lifecycle } from 'recompose'
+
+let timer = null
```

Comment:
This is a shared variable.
Every toast will share this timer, this is a bad pattern to use.

Instead, you should init this variable into your hoc, by example :
```es6
lifecycle({
  componentDidUpdate(prevProps) {
     this.timer = addDelay(prevProps, this.props, defaultDelay, this.timer)
  },
  componentWillUnMount() {
    clearTimeout(this.timer)
  }
})
```